addons/osio-addon/src/main/java/io/fabric8/launcher/osio/tenant/TenantRequests.java

```diff
@@ -0,0 +1,58 @@
+package io.fabric8.launcher.osio.tenant;
+
+import javax.ws.rs.core.HttpHeaders;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.launcher.base.identity.IdentityHelper;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.osio.EnvironmentVariables;
+import okhttp3.Request;
+
+import static io.fabric8.launcher.base.http.ExternalRequest.readJson;
+
+public final class TenantRequests {
+
+    private TenantRequests() {
+        throw new IllegalAccessError("Helper class");
+    }
+
+    public static Tenant getTenant(final TokenIdentity osioToken) {
```

Comment:
Maybe move this code to `TenantProducer`?

Reply:
same here as the other, I really like this principle :)
https://en.wikipedia.org/wiki/Single_responsibility_principle